photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditor.java

```diff
@@ -64,10 +98,62 @@ private PhotoEditor(Builder builder) {
         this.isTextPinchZoomable = builder.isTextPinchZoomable;
         this.mDefaultTextTypeface = builder.textTypeface;
         this.mDefaultEmojiTypeface = builder.emojiTypeface;
+        this.viewState = new ViewState();
         mLayoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
         brushDrawingView.setBrushViewChangeListener(this);
-        addedViews = new ArrayList<>();
-        redoViews = new ArrayList<>();
+
+
+        // A listener for the image view that helps with the focus view logic.
+        // i.e when you press on an empty space without stickers, it will de-select the focused sticker.
+        class ImageViewListener extends GestureDetector.SimpleOnGestureListener {
```

Comment:
We also need to move out this class passing the requires object through constructor.

Reply:
Done